### PR TITLE
API Permission Level

### PR DESCRIPTION
**Pull request by SoSorryTT**
## PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Others. (Please describe)

### Changes

1. Now only the author and subscriber of the Tag and Assignment can interact with their objects.
2. When create assignment or tag now auto fill author.

### Fixes

1. Fix the tag author bug that can not fill author. #26

### CODE REVIEW

File: garnbarn_api/views.py [hunk 1]
```diff
@@ -82,7 +87,8 @@ class TagViewset(viewsets.ModelViewSet):
     serializer_class = TagSerializer
 
     def get_queryset(self):
-        tag = Tag.objects.get_queryset().order_by('id')
+        user_data = self.request.user.get_json_data()["uid"]
```
Comment (sirateek) [inline garnbarn_api/views.py:93]: Maybe this can be minimized by `self.request.user.uid`

File: garnbarn_api/views.py [hunk 2]
```diff
@@ -106,7 +112,8 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.data, status=status.HTTP_200_OK)
 
     def perform_create(self, serializer):
-        serializer.save()
+        user_data = self.request.user.get_json_data()["uid"]
```
Comment (sirateek) [inline garnbarn_api/views.py:118]: Maybe this can be minimized by `self.request.user.uid`

File: garnbarn_api/views.py [hunk 3]
```diff
@@ -16,12 +18,14 @@ class AssignmentViewset(viewsets.ModelViewSet):
     serializer_class = AssignmentSerializer
 
     def get_queryset(self):
+        user_data = self.request.user.get_json_data()["uid"]
```
Comment (sirateek) [inline garnbarn_api/views.py:24]: Maybe this can be minimized by `self.request.user.uid`

File: garnbarn_api/views.py (hunk 1 above)
Comment (SoSorryTT) [reply garnbarn_api/views.py]: Done in commit bc03e83.

File: garnbarn_api/views.py (hunk 2 above)
Comment (SoSorryTT) [reply garnbarn_api/views.py]: Done in commit bc03e83.

File: garnbarn_api/views.py (hunk 3 above)
Comment (SoSorryTT) [reply garnbarn_api/views.py]: Done in commit bc03e83.

**Review by sirateek — APPROVED** [review]
LGTM

**Pull request merged**